### PR TITLE
test: Add tests for duplicate private methods (early-errors)

### PR DESCRIPTION
**Pull request by mkubilayk**
Adds tests for duplicate private methods according to the test plan posted in #1343.

|Valid|Case|Description|
|:---:|:---|:---|
| :red_circle: | `grammar-privatemeth-duplicate-meth-meth` | Cannot have multiple private methods with the same name |
| :red_circle: | `grammar-privatemeth-duplicate-meth-get` | Cannot have a private method and a private getter with the same name |
| :red_circle: | `grammar-privatemeth-duplicate-meth-set ` | Cannot have a private method and a private setter with the same name |
| :red_circle: | `grammar-privatemeth-duplicate-get-get` | Cannot have multiple private getters with the same name |
| :red_circle: | `grammar-privatemeth-duplicate-set-set` | Cannot have multiple private setters with the same name |
| :white_circle: | `grammar-privatemeth-duplicate-get-set` | Can have a private getter and a private setter with the same name |
| :red_circle: | `grammar-privatemeth-duplicate-meth-field` | Cannot have a private method and a private field with the same name |
| :red_circle: | `grammar-privatemeth-duplicate-get-field` | Cannot have a private getter and a private field with the same name |
| :red_circle: | `grammar-privatemeth-duplicate-set-field` | Cannot have a private setter and a private field with the same name |
| :red_circle: | `grammar-privatemeth-duplicate-meth-staticmeth` | Cannot have a private method and a private static method with the same name |
| :red_circle: | `grammar-privatemeth-duplicate-meth-staticfield` | Cannot have a private method and a private static field with the same name |
| :white_circle: | `grammar-privatemeth-duplicate-meth-nestedclassmeth` | Nested class can shadow private methods  |

---

@littledan Could you please have a look and see if I missed anything? One thing I wasn't clear is whether we can have a private getter/setter and a private field with the same name. 

```js
class C {
   #x;

   get #x() {} 
}
```

I can add a test for this if you tell me what the correct behavior should be.

---
Tracking Issue: #1343
cc: @jbhoosreddy

### CODE REVIEW

**Comment by littledan** [comment]
Yes, that last sample would be invalid as well. Is there anything about the specification that makes this ambiguous?

**Comment by mkubilayk** [comment]
On a second look, it is clear in the spec. My bad. I will add a test for that. Thanks for the answer!

**Comment by mkubilayk** [comment]
Added tests for that.

**Comment by jbhoosreddy** [comment]
I think in the contributing docs it says we should not PR the generated tests

**Comment by mkubilayk** [comment]
You are right. However, #1761 includes them. 

@leobalter @rwaldron: Should I remove generated tests?

File: src/class-elements/grammar-privatemeth-duplicate-meth-staticfield.case [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2018 Kubilay Kahveci (Bloomberg LP). All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+desc: It's a SyntaxError if a class contains a private method and a private static field with the same name
```
[the remark this answers is not in the document]
Comment (rwaldron) [reply src/class-elements/grammar-privatemeth-duplicate-meth-staticfield.case]: https://github.com/tc39/proposal-private-methods/issues/46

**Comment by rwaldron** [comment]
> Should I remove generated tests?

Yes, generate the tests after the PR is approved.

**Comment by mkubilayk** [comment]
@rwaldron thanks for the review. I removed generated tests and added missing parameters for the setters.

File: src/class-elements/grammar-privatemeth-duplicate-meth-staticfield.case (hunk 1 above)
[the remark this answers is not in the document]
Comment (rwaldron) [reply src/class-elements/grammar-privatemeth-duplicate-meth-staticfield.case]: Ok, I've verified that this is intentional here: https://github.com/tc39/proposal-private-methods/issues/46

**Comment by rwaldron** [comment]
Ok, now that review is complete and approved, please squash the first four commits (you can squash into as many commits as you'd like, but I think 2 makes sense here), then generate the tests and make a commit with just the generated tests. I will merge as soon as I can!

**Comment by mkubilayk** [comment]
@rwaldron done. Thanks again!

**Comment by rwaldron** [comment]
Excellent, thanks!

**Pull request merged**